### PR TITLE
feat(lh-86968): add terraform resource to add user groups to MSP-managed tenant

### PR DESCRIPTION
**Pull request by siddhuwarrier**
https://jira-eng-rtp3.cisco.com/jira/browse/LH-86968

### Description

Allow an MSP admin to use terraform to add user groups to a tenant managed by the MSP portal. TODO:
acceptance tests

### CODE REVIEW

File: client/internal/url/url.go [hunk 1]
```diff
@@ -237,3 +237,15 @@ func DeleteUsersInMspManagedTenant(baseUrl string, tenantUid string) string {
...
+
+func CreateUserGroupsInMspManagedTenant(baseUrl string, tenantUid string) string {
+	return fmt.Sprintf("%s/api/rest/v1/msp/tenants/%s/users/groups", baseUrl, tenantUid)
+}
+
+func GetUserGroupsInMspManagedTenant(baseUrl string, tenantUid string, limit int, offset int) string {
+	return fmt.Sprintf("%s/api/rest/v1/msp/tenants/%s/users/groups?limit=%d&offset=%d", baseUrl, tenantUid, limit, offset)
+}
+
+func DeleteUserGroupsInMspManagedTenant(baseUrl string, tenantUid string) string {
```
[the remark this answers is not in the document]
Comment (siddhuwarrier) [reply client/internal/url/url.go:249]: It isn't deleting all user groups though, it's only deleting the user groups that the user has added using Terraform.

File: client/client.go [hunk 2]
```diff
@@ -311,3 +312,15 @@ func (c *Client) GenerateApiTokenForUserInMspManagedTenant(ctx context.Context,
...
+
+func (c *Client) CreateUserGroupsInMspManagedTenant(ctx context.Context, tenantUid string, userGroups *[]usergroups.MspManagedUserGroupInput) (*[]usergroups.MspManagedUserGroup, *usergroups.CreateError) {
+	return usergroups.Create(ctx, c.client, tenantUid, userGroups)
+}
+
+func (c *Client) ReadUserGroupsInMspManagedTenant(ctx context.Context, tenantUid string, userGroups *[]usergroups.MspManagedUserGroupInput) (*[]usergroups.MspManagedUserGroup, error) {
+	return usergroups.ReadCreatedUserGroupsInTenant(ctx, c.client, tenantUid, userGroups)
+}
+
+func (c *Client) DeleteUserGroupsInMspManagedTenant(ctx context.Context, tenantUid string, deleteInput *usergroups.MspManagedUserGroupDeleteInput) (interface{}, error) {
```
[the remark this answers is not in the document]
Comment (siddhuwarrier) [reply client/client.go:324]: It isn't deleting all user groups though, it's only deleting the user groups that the user has added using Terraform.

File: client/msp/usergroups/read.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+		readUserGroupDetailsMap[createdUserGroup.GroupIdentifier] = MspManagedUserGroup{
+			GroupIdentifier: createdUserGroup.GroupIdentifier,
+			IssuerUrl:       createdUserGroup.IssuerUrl,
+			Role:            createdUserGroup.Role,
+			Name:            createdUserGroup.Name,
+			Notes:           createdUserGroup.Notes,
+		}
+	}
+
+	limit := 200
```
[the remark this answers is not in the document]
Comment (siddhuwarrier) [reply client/msp/usergroups/read.go:25]: So this isn't a parameter. I'm looping through the full list of user groups to find the ones that we need to remove.

File: client/msp/usergroups/read.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	offset := 0
+	count := 1
+	var readUrl string
+	var userGroupPage MspManagedUserGroupPage
+	foundUsernames := mapset.NewSet[string]()
+
+	for count > offset {
```
[the remark this answers is not in the document]
Comment (siddhuwarrier) [reply client/msp/usergroups/read.go:32]: I want to loop through the list until we get the full list of items.

File: client/msp/usergroups/read.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (talhazi) [reply client/msp/usergroups/read.go:25]: ah now I get it, cool

**Pull request merged**